Modules/_threadmodule.c

```diff
@@ -1137,13 +1137,8 @@ information about locks.");
 static PyObject *
 thread_get_ident(PyObject *self)
 {
-    long ident;
-    ident = PyThread_get_thread_ident();
-    if (ident == -1) {
-        PyErr_SetString(ThreadError, "no current thread ident");
-        return NULL;
-    }
-    return PyLong_FromLong(ident);
+    unsigned long ident = PyThread_get_thread_ident();
```

Comment:
Good catch. Don't know why I removed the check here. The code was initially written 4 years ago.